crates/notion-core/src/tool/mod.rs

```diff
@@ -442,4 +499,67 @@ mod tests {
             );
         }
     }
+
+    mod from_strings {
```

Comment:
Should also have assertions for the success cases that the number of elements in the result vector matches the number in the input vector.

Reply:
Agreed!